### ISSUE TITLE
1.3.1 and 1.4.1 understanding documents may be contrary  on the use of color

### ISSUE THREAD

**Issue by mraccess77** [w3c/wcag#2239]
SC 1.4.1 understanding doc indicates that it only applies to the visual presentation of color and that 1.3 covers programmatic exposure of the information communicated by color.  SC 1.3.1's understanding doc indicates that color is addressed by SC 1.4.1.  So which is it?

[Success Criterion 1.4.1 ](https://www.w3.org/WAI/WCAG21/Understanding/use-of-color)addresses the specific case of color, rather than Success Criterion 1.3.1.
---
Other forms of perception are covered in [Guideline 1.3](https://www.w3.org/TR/WCAG21/#adaptable) including programmatic access to color and other visual presentation coding.

**Comment by patrickhlauke** [comment]
Ah yes, seems to be a discrepancy here between 1.4.1 https://w3c.github.io/wcag/understanding/use-of-color.html

> This success criterion addresses color perception specifically. Other forms of perception are covered in [Guideline 1.3](https://w3c.github.io/wcag/guidelines/#adaptable) including programmatic access to color and other visual presentation coding.

And then the note in 1.3.1 understanding https://w3c.github.io/wcag/understanding/info-and-relationships.html

> It is not required that color values be programmatically determined. The information conveyed by color cannot be adequately presented simply by exposing the value. Therefore, [Success Criterion 1.4.1 ](https://w3c.github.io/wcag/understanding/use-of-color)addresses the specific case of color, rather than Success Criterion 1.3.1.

I wonder if the fix here is to change the note for 1.4.1 to

> This success criterion addresses color perception specifically. Other forms of perception (including other aspects of visual presentation) are covered in [Guideline 1.3](https://w3c.github.io/wcag/guidelines/#adaptable)

**Comment by patrickhlauke** [comment]
> In this case, rather the note at 1.3.1 needs to be changed. 1.3.1 should then not require that color values be transmitted, but that the information representing the colors be transmitted programmatically (e.g., for blind people or people who use high contrast mode). 

Which is what the note in 1.3.1 already says, no (emphasis mine)? https://w3c.github.io/wcag/understanding/info-and-relationships.html

> *It is not required that color values be programmatically determined*. The information conveyed by color cannot be adequately presented simply by exposing the value. Therefore, [Success Criterion 1.4.1 ](https://w3c.github.io/wcag/understanding/use-of-color)addresses the specific case of color, rather than Success Criterion 1.3.1.

**Comment by JAWS-test** [comment]
> It is not required that color values be programmatically determined. The information conveyed by color cannot be adequately presented simply by exposing the value. Therefore, [Success Criterion 1.4.1 ](https://w3c.github.io/wcag/understanding/use-of-color)addresses the specific case of color, rather than Success Criterion 1.3.1.

should be changed to

> It is not required that color values be programmatically determined. The information conveyed by color cannot be adequately presented simply by exposing the value. Therefore, information conveyed by means of color should be conveyed programmatically or in text form. Furthermore, it should be noted that 1.4.1 defines additional requirements for color. 

Reason: 1.3.1 suggests not being responsible for color. But this is not true, 1.3.1 is relevant for color coded information. 

Alternative: Removing the note would also make sense, because we don't write: "For headings, the information cannot be adequately conveyed by conveying font size, color, and style." Also headings are conveyed programmatically with h1 to h6 and not via their visual properties (e.g. bold and 20px instead of 16px)

**Comment by bruce-usab** [comment]
I like the edit to Understanding proposed by @JAWS-test and it addresses the discrepancy noted by @mraccess77 in OP.

**Comment by patrickhlauke** [comment]
i'd say the note in 1.4.1 is also misleading still ... as it talks about programmatic access to color, which is then handwaved away once you're in 1.3.1.

maybe also change

> This success criterion addresses color perception specifically. Other forms of perception are covered in [Guideline 1.3](https://w3c.github.io/wcag/guidelines/#adaptable) including programmatic access to color and other visual presentation coding.

to something like 

> This success criterion addresses color perception specifically. Other forms of perception are covered in [Guideline 1.3](https://w3c.github.io/wcag/guidelines/#adaptable), including information conveyed by color in general and other aspect of visual presentation

**Comment by awkawk** [comment]
> It is complicated by the fact that F13 talks about alternative text for SC 1.4.1. https://www.w3.org/TR/2008/WD-WCAG20-TECHS-20081103/F13

@mraccess77 that is an old version of F13, the current version is 1.1.1 and 1.4.1: https://www.w3.org/TR/WCAG20-TECHS/F13.html

**Comment by mraccess77** [comment]
Sorry for the wrong link for F13- but my comment still applies as the current version (which I did read prior to posting) implies that you can meet 1.4.1 by updating the alternative text for an image that uses color to communicate meaning.   If you only updated the alt text which is not visible then you could only meet SC 1.1.1 and not SC 1.4.1.

**Comment by awkawk** [comment]
@mraccess77 I wasn't thinking about that once I saw the 1.1.1+1.4.1, but you are completely right.

Rethinking my earlier statement. I think that I was too hasty in stating that 1.3.1 covered some color situations. Currently thinking:
1. It looks like I screwed up a WG errata - The CFC from 2017 (https://lists.w3.org/Archives/Public/w3c-wai-gl/2017AprJun/0786.html) indicates that the WG agreed that the notes for 1.4.1 and 1.3.3 would be removed for clarity. Yikes. This should have been reflected in WCAG 2.1, but wasn't. @alastc, sorry!
2. F13 needs to be changed as it does have problems with 1.4.1 as pointed out.
4. Clean up of color on 1.4.1 and 1.3.1 Understanding docs.

**Comment by bruce-usab** [comment]
I agree with cleanup noted.  While F13 probably is not a good example, I think that 1.3.1 covers _some_ color situation.  For example:  <q>important terms are circled and highlighted in yellow</q> meets 1.4.1 (since the circles are non-color visual means of conveying information</q> but could fail 1.3.1 (e.g., the CSS effects are via `span` instead of `strong`).

Also, I will mention that I remember that exposing the hex value (for things like bar charts, not color pallets) was an unhelpful tread  in the early days of WCAG 2.0.  I think that bad tread influenced the current wording in Understanding.

**Comment by mraccess77** [comment]
I'm not sure that test processes like Trusted Tester have a good place to fail color related issues that are non-visual as the tests for SC 1.4.1 only cover visual differentiation of use of color.    We have a sufficient technique under G138 but I feel like many testers have historically failed use of color broadly under SC 1.4.1.

**Comment by patrickhlauke** [comment]
i might be wrong/oversimplifying, but to me:

* 1.3.1 is really broadly "anything that's conveyed visually (and has meaning/purpose) is also conveyed in structure or text somehow".
* 1.4.1 is more narrowly "it relies on user's ability to differentiate colours (hues)"

if Trusted Tester etc don't make this sort of distinction / have appropriate tests, then ... they may need to be changed to better reflect this?

**Comment by mraccess77** [comment]
SC 1.4.1 is only written to visually distinguish color for people who can see.  it doesn't appear to be written for people who cannot see. "Color is not used as the only visual means of conveying information"

**Comment by patrickhlauke** [comment]
> SC 1.4.1 is only written to visually distinguish color for people who can see. it doesn't appear to be written for people who cannot see. "Color is not used as the only visual means of conveying information". Keyword "visual".

we agree on this (but i didn't make it crystal clear in my simplification)

**Comment by bruce-usab** [comment]
Anyone on this thread have an idea why Normative label on this issue?
